### PR TITLE
Deltasync: Add growth/shrink unittest for downloads

### PR DESCRIPTION
**Pull request by ckamm**
Similar to the one for uploads, to test these corner cases more.

@ahmedammar

### CODE REVIEW

File: test/testzsync.cpp [hunk 1]
```diff
@@ -42,12 +42,17 @@ private slots:
         SyncOptions opt;
         opt._deltaSyncEnabled = true;
         opt._deltaSyncMinFileSize = 0;
+        // Reduce the chunk size, allowing us to work with smaller data
+        int chunkSize = 2 * 1000 * 1001;
```
Comment (ogoffart) [inline test/testzsync.cpp]: Note: since we build with C++14,  we can also use the group separator notation: `2'002'000`  It might actually be more readable.

File: test/testzsync.cpp [hunk 2]
```diff
@@ -42,12 +42,17 @@ private slots:
         SyncOptions opt;
         opt._deltaSyncEnabled = true;
         opt._deltaSyncMinFileSize = 0;
+        // Reduce the chunk size, allowing us to work with smaller data
```
Comment (ogoffart) [inline test/testzsync.cpp]: Fine as long as there are other test testing with the actual chunk size.

File: test/testzsync.cpp (hunk 1 above)
Comment (guruz) [reply test/testzsync.cpp]: might also be nice in future to name variables with a unit.
E.g. chunkSizeBytes, timeOutMsec
